packages/clay-core/src/tree-view/useTree.ts

```diff
@@ -126,11 +158,24 @@ type PatchAdd = {
 	value: unknown;
 };
 
+// Operation of `remove` value to the document structure.
+// RFC 6902 (JSON Patch) 4.2
+type PatchRemove = {
+	op: 'remove';
+	path: Array<number>;
+};
+
+type PatchRename = {
+	name: string;
+	op: 'rename';
+	path: Array<number>;
+};
```

Comment:
@julien we could instead of using the `rename` naming we can go with `replace` which follows the same RFC pattern in section 4.3 and also `replace` sounds more like an immutability oriented pattern. What do you think?